### PR TITLE
Complete HW1

### PR DESCRIPTION
**Pull request by MinhDucMa**
###Code changes I mad
For this HW1, I may change on `01-selectors-md` to answer and pass all level from the CSSDIner game which allow player to type in suitable selectors to correctly select the required element from HTML. 

In addition, I also implemented several changes on both file `02-palindrome.html` and `02-palindrome.js`. These implementation include update the UI page of `02 - Palindrome` to make it precisely reflect the example provided for this exercise. Then, which those implementation, the webpage now able to accept user input and check if  the input is palindrome or not. If input is palindrome, the webpage will display a green text to notice that it is a palindrome. Otherwise, the text will be red for input is not a palindrome.

###Why I made these changes
From the implementation for the second exercise, I utilized bootstrap to address the correct sizing and layout of element as well as the background color. Moreover, I also implemented a JavaScript file to allow the webpage accepts user input and response with corresponding answer.

##What feedback I am looking for
During the implementation process, I have attempted to adjust the `padding-left` element of the `div` that contains all element aiming for 100% match with the example. However, despite trying different methods, I still unable to produce the desired result. Therefore, any feedbacks will be appreciated to help me improve my code quality and performance as well as address the padding issue that I am having

### CODE REVIEW

File: hw1/02-palindrome.html [hunk 1]
```diff
@@ -6,22 +6,29 @@
...
         <div>
-          <label>Enter a positive number:</label>
-          <input type="number" />
-        </div>
-        <div>
-          <p>Is this a palindrome?</p>
-          <div></div>
+          <h1>Exercise 02 - Palindrome</h1>
+          <div class="mt-4 mb-4">
+            <label>Enter a positive number:</label>
```
Comment (CristianTS599) [inline hw1/02-palindrome.html]: This label tag and the following input tag as they are will trigger a Accessibility error. your label needs the "for" attribute set to the ID value of the input tag (the input tag needs a ID attribute.

File: hw1/02-palindrome.js [hunk 2]
```diff
@@ -1,3 +1,19 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
 
-elem.addEventListener('input', handleInput);
+elem.addEventListener("input", handleInput);
```
Comment (CristianTS599) [inline hw1/02-palindrome.js:3]: I looks like you stuck with double quotes for strings all in your file this is good I like the consistency!

Comment (CristianTS599) [inline hw1/02-palindrome.js:3]: Looks like all your strings use double quotes, this is good! I like the consistency.

File: hw1/02-palindrome.js [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+
+function handleInput() {
+  const input = document.querySelector("input").value;
+  const result = document.getElementById("result");
+
+  let inputReversed = input.split("").reverse().join("");
+  if (inputReversed === input) {
+    document.getElementById("result").innerHTML = "Yes, this is a palindrome";
```
Comment (CristianTS599) [inline hw1/02-palindrome.js]: On line 7 you have "const result = document.getElementById("result");" here on line 11 and 15 you could replace the "document.getElementById("result").innerHTML..." with "result.innerHTML" it saves you the work of extracting that element three times down to just once. Other than that Great job I ran your page and it seems to work great!

**Review by CristianTS599 — COMMENTED** [review]
Hey Minh Ma, I completed your code review and I left a few comments as suggestions. if you have any questions about it let me know! but other than those comments I ran your page and it seems to work great good job! 

In terms of the padding for your main div I got padding to work by using the [Bootstrap 5 documentation](https://getbootstrap.com/docs/5.0/utilities/spacing/) for example adding a "ps-5" class will give your div a big padding on the left side take a look and let me know if this doesn't solve your issue and maybe I can attempt to help some more!

**Comment by 2miles** [comment]
First thing I noticed was that the styling and Bootstrap looked great. Looks exactly like the example. I think I might of messed up on mine because I didn't realize we maybe were supposed to mimic her example styling. I just did my own thing with the styling. I'll probable go back and try to make it look just like hers.

After running the code one bug I found was that after entering any non-numeric characters into the text box the output says 'Yes this is a palindrome'. Also after entering a palindrome then continuing to append non-numeric characters the output continues to say 'Yes this is a palindrome'. 

Running WAVE on the page result in a contrast error with the red and green text against that background. Just changing the colors to 'darkred' and 'darkgreen' respectively fixed that for me.

File: hw1/02-palindrome.html (hunk 1 above)
Comment (MinhDucMa) [reply hw1/02-palindrome.html]: Hi Cristian, thank you for all the feedback! I completely forgot about handling the Accessibility aspect. In addition, I did try to add the "ps-5" class before but nothing seem to change. That why I also try to add "styled" to attempt adjusting the padding with CSS but still nothing happen.  Do you think I should try anything else?

**Comment by MinhDucMa** [comment]
Just make some more changes to fix the bug that @2miles mentioned. However, relating to the suggestion that you had for the color contrast issue when you said changing the color to 'darkred' and 'darkgreen', did you mean using the 'styled' property to directly add CSS color? Or there are some bootstrap classes that can help to achieve that?

**Comment by 2miles** [comment]
@MinhDucMa 
There are multiple ways of doing it. I ended up doing something like this:
```js
let element = document.getElementById(' element_id ');
element.style.color = colorCode;
```
`note:` when assigning a color like this I think you need to use the camelCase versions of the color code.
```js 
'darkRed' 'darkGreen'
```

**Comment by devonfox** [comment]
image

Nice work Minh! Everything looks great, and I noticed you used the bootstrap classes for the colors, great work. I did notice on Firefox there is a bug when you enter characters in the input field.  On Chrome, you are unable to do this, but they allow it on Firefox.  Prof. Paun mentioned that this is something we will want to check for.  You can use isNaN or just check if the string is empty.  I just checked if the string is empty, as in Firefox when a number input contains a non-numeric character, `input.event.value` will contain a an empty string "".  Hope that helps!  Let me know if you need clarification.

**Pull request merged**